packages/noports_core/lib/src/sshnp/impl/sshnp_dart_pure_impl.dart

```diff
@@ -2,14 +2,35 @@ import 'dart:async';
 
 import 'package:at_client/at_client.dart';
 import 'package:dartssh2/dartssh2.dart';
+import 'package:noports_core/src/sshnp/impl/notification_request_message.dart';
 import 'package:noports_core/sshnp_foundation.dart';
 
 class SshnpDartPureImpl extends SshnpCore
     with SshnpDartSshKeyHandler, DartSshSessionHandler {
-  SshnpDartPureImpl(
-      {required super.atClient,
-      required super.params,
-      required AtSshKeyPair? identityKeyPair}) {
+  SshnpDartPureImpl({
+    required super.atClient,
+    required super.params,
+    required AtSshKeyPair? identityKeyPair,
+  }) {
+    // TODO Defensive code to prevent use of rvd auth and rv traffic encryption
+    // TODO until they have been properly implemented with an in-memory RV.
+    // TODO At that time, make these four params "final" again
+    if (params.discoverDaemonFeatures) {
+      logger.shout('$runtimeType: disabling discoverDaemonFeatures flag');
+      params.discoverDaemonFeatures = false;
+    }
+    if (params.encryptRvdTraffic) {
+      logger.shout('$runtimeType: disabling encryptRvdTraffic flag');
+      params.encryptRvdTraffic = false;
+    }
+    if (params.authenticateDeviceToRvd) {
+      logger.shout('$runtimeType: disabling authenticateDeviceToRvd flag');
+      params.authenticateDeviceToRvd = false;
+    }
+    if (params.authenticateClientToRvd) {
+      logger.shout('$runtimeType: disabling authenticateClientToRvd flag');
+      params.authenticateClientToRvd = false;
+    }
```

Comment:
As mentioned in one of my other comments, it would be useful to have an in-memory RV with the openssh client as well, then we can call the ssh command as a child process using Process.run()